include/oneapi/dpl/pstl/iterator_defs.h

```diff
@@ -31,23 +31,18 @@ namespace __internal
 // SFINAE with a non-iterator type by providing a default value.
 template <typename _IteratorTag, typename... _IteratorTypes>
 auto
-__is_iterator_of(int) -> decltype(
-    ::std::conjunction<::std::is_base_of<
-        _IteratorTag, typename ::std::iterator_traits<::std::decay_t<_IteratorTypes>>::iterator_category>...>{});
+__is_iterator_of(int) -> typename ::std::conjunction<::std::is_base_of<
```

Comment:
Here the returning type is the resulting type of `std::conjunction` which is `std::true_type` or `std::false_type` (according to ).

Reply:
If I'm understanding the problem correctly, `std::conjunction` falls into the same issue as doing `struct __is_random_access_iterator`.  There may be a simpler way of doing this, but to end up with exactly `std::true_type` or `std::false_type`, I think this may need to be `std::integral_constant<bool, std::conjunction_v<...>>` where `...` is the list of `std::true_type`, `std::false_type` for the individual iterators provided.

Reply:
Never mind.  I missed the `::type` at the end of your formulation which is, in fact, the more direct / simpler way I thought might exist. :)